crates/pathfinder/src/storage/contract.rs

```diff
@@ -0,0 +1,132 @@
+use crate::{
+    core::{ContractAddress, ContractHash},
+    storage::{DB_VERSION_CURRENT, DB_VERSION_EMPTY},
+};
+
+use pedersen::StarkHash;
+use rusqlite::{named_params, OptionalExtension, Transaction};
+
+/// Migrates the [ContractsTable] to the [current version](DB_VERSION_CURRENT).
+pub fn migrate(transaction: &Transaction, from_version: u32) -> anyhow::Result<()> {
+    ContractsTable::migrate(transaction, from_version)
+}
+
+/// Stores StarkNet contract information, specifically a contract's
+///
+/// - [address](ContractAddress)
+/// - [hash](ContractHash)
+/// - byte code
+/// - ABI
+/// - definition
+pub struct ContractsTable {}
+
+impl ContractsTable {
+    /// Migrates the [ContractsTable] from the given version to [DB_VERSION_CURRENT].
+    fn migrate(transaction: &Transaction, from_version: u32) -> anyhow::Result<()> {
+        match from_version {
+            DB_VERSION_EMPTY => {} // Fresh database, continue to create table.
+            DB_VERSION_CURRENT => return Ok(()), // Table is already correct.
+            other => anyhow::bail!("Unknown database version: {}", other),
+        }
+
+        transaction.execute(
+            r"CREATE TABLE contracts (
+                    address    BLOB PRIMARY KEY,
+                    hash       BLOB NOT NULL,
+                    code       BLOB,
+                    abi        BLOB,
+                    definition BLOB
```

Comment:
Just for clarity -  definition as in compiled contract code/contract definition in python code?

Reply:
The exact structure is unknown until we get the api update from starkware side. I just took what we currently have (code + abi) and added definition which we are lacking.